routing/index_road_graph.cpp

```diff
@@ -97,10 +97,16 @@ void IndexRoadGraph::GetEdges(Junction const & junction, bool isOutgoing, TEdgeV
   }
 }
 
+m2::PointD IndexRoadGraph::GetJunctionPoint(Segment const & segment, bool front) const
+{
+  return front && m_starter.FitsFinish(segment) ? m_starter.GetFinishVertex().GetPoint()
```

Comment:
It doesn't required by ReconstructPath() right now but I guess it's worth doing so. Pls see discussion below.